lib/widgets/layout/base_layout.dart

```diff
@@ -0,0 +1,105 @@
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+
+import 'package:myapp/src/router/coordinator.dart';
+import 'package:myapp/src/theme/styles.dart';
+import 'package:myapp/widgets/button/custom_button.dart';
+
+class BaseLayout extends StatelessWidget {
+  final String buttonText;
+  final VoidCallback onButtonPressed;
+  final Widget content;
+  final String? currentView;
+
+  const BaseLayout({
+    super.key,
+    required this.buttonText,
+    required this.onButtonPressed,
+    required this.content,
+    this.currentView,
+  });
+
+  void _onSignInPressed() {
```

Comment:
Do not do the specific logic into a base class => use onTapPress callback and do the logic where the instance declared